src/js_object_accessor.hpp

```diff
@@ -68,10 +68,10 @@ class NativeAccessor {
 
     OptionalValue value_for_property(ValueType dict, std::string const& prop_name, size_t prop_index) {
         ObjectType object = Value::validated_to_object(m_ctx, dict);
-        if (!Object::has_property(m_ctx, object, prop_name)) {
+        ValueType value = Object::get_property(m_ctx, object, prop_name);
```

Comment:
We could easily apply this pattern many other places 😄

Reply:
But now we cannot know if we have the property or the value in `undefined`.

Reply:
Drawing a distinction between no value and `undefined` is pretty strange for JS to begin with, and most of the API already didn't. I don't think this is a behavioral change to any existing working code; it's purely a widening in what sorts of objects we accept to create().